Source/DataModels/Resources/ResourceMaterial.h

```diff
@@ -30,6 +30,7 @@ class ResourceMaterial : virtual public Resource
 	const std::shared_ptr<ResourceTexture>& GetOcclusion() const;
 	const std::shared_ptr<ResourceTexture>& GetMetallic() const;
 	const std::shared_ptr<ResourceTexture>& GetSpecular() const;
+	const std::shared_ptr<ResourceTexture>& GetEmission() const;
```

Comment:
I'd return by copy so that the reference counter is increased.
```suggestion
	std::shared_ptr<ResourceTexture> GetEmission() const;
```
Same for all of the above, if you don't mind.